### PR TITLE
feat: improved session serialization debug tool

### LINKED ISSUE #45: Add more information to debug tool output
To debug Java serialization issues, the Java platform offers the `sun.io.serialization.extendedDebugInfo` system property that enables logging of the serialization stack, providing helpful information about objects and fields being serialized.

In the context of the Kit Debug Tool, a suggestion was to automatically set the property, but this may not work well for the purposes of the tool.

Reasons to not apply the property automatically are:

- the flag is read, and its value is stored in a static member of `ObjectOutputStream` class.
  This means that, if the property is not set when the JVM starts and `ObjectOutputStream` is used before the Debug Tool, changing it has no effects because the value has already been read
- When the property is enabled, the stack is printed only if there's a serialization error.
  The tool aims to discover as many errors as possible with each run, and for this reason, it tracks and replaces unserializable classes with null values, so no stack information will be printed. If the tool does not replace the invalid values, it will provide not value over than simple adding the property.
- When storing the stack trace, the `toString()` method is used to represent objects
  If the method fails for any reason (poor written or causing side effects like Hibernate `PersistentList`) the serialization process will fail, but not because the objects are not serializable.
- The property is used only for serialization. It has no effect during deserialization when the weirdest exception may arise (e.g. SerializedLambda class cast exception)

However, the debug tool can anyway try to profit of the information available when the property is enabled,
but we may require the user to:
- manually enable the property at JVM start
- make reflection access to `ObjectOutputStream` and `ObjectInputStream`  possible by adding the `--add-opens java.base/java.io=ALL-UNNAMED` instruction

With the above configuration, we will be able to read the current stack when inspecting unserializable objects and then print the gathered information.

For deserialization, it would be helpful to track every object, associate the trace and print it if deserialization fails.
However, tweaking deserialization process may be hard because hooks are available only once an object is completely deserialized, and exceptions may have already been thrown.

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

The tool now works only with --add-opens and extendedDebugInfo system property.
Unserializable objects are replaced by null and restored during deserialization to avoid NPEs.
Object with throwing toString are replaced with an object capable to serialize fields and regenerate the original object type on deserialization.

Fixes #45

## Type of change

- [ ] Bugfix
- [X] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by mcollovati** [comment]
This draft PR is a placeholder for session serialization debug tool improvements.
Code needs to be cleaned and simplified.

File: kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/SerializationDebugRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+                    .getAttribute(SERIALIZATION_TEST_REQUEST_ATTRIBUTE_KEY);
+            if (action instanceof Runner) {
+                LOGGER.debug(
+                        "Vaadin Request processed, Running Session Serialized Debug Tool");
+                ((Runner) action).accept(req);
+            }
+        }
+    }
+
+    public static void serializeAndDeserialized(WrappedSession session,
```
Comment (MatthewVaadin) [inline kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/SerializationDebugRequestHandler.java:185]: Rename method to `serializeAndDeserialize`.

File: kubernetes-kit-starter/src/test/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/FailingToStringReplacerTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        Map<String, Object> map = Map.of("TEST", 123);
+
+        @Override
+        public String toString() {
+            return "Booom! " + x.length();
+        }
+    }
+
+}
```
Comment (MatthewVaadin) [inline kubernetes-kit-starter/src/test/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/FailingToStringReplacerTest.java:60]: Add new line at the end of file.

**Review by MatthewVaadin — CHANGES_REQUESTED** [review]

File: kubernetes-kit-starter/src/test/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/FailingToStringReplacerTest.java (hunk 2 above)
Comment (mcollovati) [reply kubernetes-kit-starter/src/test/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/FailingToStringReplacerTest.java]: done

File: kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/SerializationDebugRequestHandler.java (hunk 1 above)
Comment (mcollovati) [reply kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/SerializationDebugRequestHandler.java]: Good catch!
Done

**Review by MatthewVaadin — APPROVED** [review]

**Pull request merged**